agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -544,14 +557,11 @@ func (s *Server) realHandleStream(streamReq HandleStreamRequest) error {
 				// At this point we have a valid ResourceURL and we are subscribed to it.
 
 				switch {
-				case req.ResponseNonce == "" && req.Error != nil:
-					return grpcstatus.Error(codes.InvalidArgument, "initial subscription request for a resource type must not contain an error")
-
-				case req.ResponseNonce != "" && req.Error == nil: // ACK
+				case req.Error == nil: // ACK
 					// TODO(peering): handle ACK fully
 					status.TrackAck()
 
-				case req.ResponseNonce != "" && req.Error != nil: // NACK
+				case req.Error != nil: // NACK
```

Comment:
Calling out the removal of this check. Since the initial stream tracking happens in the block above, I don't think checking the ResponseNonce is necessary (correct me if I'm wrong).

The first case in particular was causing errors in tests where the initial replication request for server addresses failed (peering was deleted). The server was stuck in a loop sending replication messages.

Reply:
This seems fine to me.